process/check.py

```diff
@@ -419,3 +427,42 @@ def _process_service_check(self, name, nb_procs, bounds, tags):
             tags=service_check_tags,
             message=message_str % (status_str[status], nb_procs, name)
         )
+
+    @staticmethod
+    def add_children(parent_pids):
+        """
+        Searches children pids for each of parent
+        pids recursively.
+        :param parent_pids: set of parent pids
+        :return: set parent and children pids
+        """
+        pids = set()
+        for ppid in parent_pids:
+            try:
+                parent = psutil.Process(ppid)
+                children = parent.children(recursive=True)
+                for child in children:
+                    pids.add(child.pid)
+            except psutil.NoSuchProcess:
+                pass
+
+        return parent_pids.union(pids)
+
+    @staticmethod
+    def filter_by_user(user, pids):
+        """
+        Filter pids by it's username.
+        :param user: string with name of system user
+        :param pids: set of pids to filter
+        :return: set of filtered pids
+        """
+        filtered_pids = set()
+        for pid in pids:
+            try:
+                proc = psutil.Process(pid)
+                if proc.username() == user:
+                    filtered_pids.add(pid)
```

Comment:
Same here for a debug log line:

something like:
self.log.debug("Filtering in pid %s with username %s", pid, user)